usr/src/uts/common/brand/lx/os/lx_brand.c

```diff
@@ -2361,9 +2361,15 @@ lx_elfexec(struct vnode *vp, struct execa *uap, struct uarg *args,
 	 * Save off the important properties of the lx executable. The brand
 	 * library will ask us for this data later, when it is ready to set
 	 * things up for the lx executable.
+	 * For statically linked executable, use vaddr from Phdr.
 	 */
-	edp.ed_phdr = (uphdr_vaddr == -1) ? voffset + ehdr.e_phoff :
-	    voffset + uphdr_vaddr;
+	if (uphdr_vaddr == -1) {
+		edp.ed_phdr = voffset + ehdr.e_phoff;
```

Comment:
A comment about what uphdr_vaddr == -1 means should go here.  Oh and if uphdr_vaddr is -1, do we STILL need to check ET_EXEC and interp == NULL ?